JitsiConference.js

```diff
@@ -505,34 +536,52 @@ JitsiConference.prototype.replaceTrack = function (oldTrack, newTrack) {
             return Promise.reject(
                 new JitsiTrackError(JitsiTrackErrors.TRACK_IS_DISPOSED));
         }
-        // Set up the ssrcHandler for the new track before we add it at the lower levels
-        newTrack.ssrcHandler = function (conference, ssrcMap) {
-            if (ssrcMap[this.getMSID()]) {
-                this._setSSRC(ssrcMap[this.getMSID()]);
-                conference.room.removeListener(XMPPEvents.SENDRECV_STREAMS_CHANGED,
-                    this.ssrcHandler);
-            }
-        }.bind(newTrack, this);
-        this.room.addListener(XMPPEvents.SENDRECV_STREAMS_CHANGED,
-            newTrack.ssrcHandler);
     }
     // Now replace the stream at the lower levels
-    return this.room.replaceStream (oldTrack, newTrack)
+    return this._doReplaceTrack(oldTrack, newTrack)
         .then(() => {
             if (oldTrack) {
-                this.onTrackRemoved(oldTrack);
+                this.onLocalTrackRemoved(oldTrack);
             }
             if (newTrack) {
                 // Now handle the addition of the newTrack at the JitsiConference level
                 this._setupNewTrack(newTrack);
             }
             return Promise.resolve();
         }, (error) => {
+            if (newTrack) {
+                // NOTE It's is unclear from this method perspective on where is
```

Comment:
hmm yes, you're right. I remember it was fixing some problem, but will double check that. Maybe I messed up something when rebasing.